### DISCUSSION TITLE
Quartz.AspNetCore

### DISCUSSION THREAD

**Discussion by andrewlock** [quartznet/quartznet#910] [General] [answered]
I finally got a chance to take a look at the new Quartz.AspNetCore integration and I just have a small nitpic. The actual functionality, the `QuartzHostedSevice`, doesn't seem like it needs to be AspNetCore specific - it could easily be using the generic `Host` only right? I'm not convinced that the package should be silently adding health checks (an ASP.NET Core feature)? Apart from anything else, that's an additional NuGet dependency that would otherwise not be required.

That said, the health check _is_ a nice addition, so I wonder if it should be two separate packages, similar to the way Serilog works for example. You could have a base `Quartz.Hosting` library, which only provides the generic `Host` support, which the `Quartz.AspNetCore` depends on, and adds health checks too?

**Comment by lahma** [comment, marked as answer]
Thank you for having a look 👍 I was also a bit on the fence with this as the new hosting model doesn't depend on ASP.NET Core anymore. I'm not sure if I can introduce the new package and the dependency chain in 3.1, though users would see same functionality as the hosting would come as transitive.

I guess `Quartz.Extensions.Hosting` would be logical package name like we have `Quartz.Extensions.DependencyInjection` now for the DI stuff (following the MS naming)?

**Reply by andrewlock** [reply]
>I'm not sure if I can introduce the new package and the dependency chain in 3.1, though users would see same functionality as the hosting would come as transitive. 

Yeah, I think the fact it would be transitive would mean minimal impact on consumers, but still probably worth pushing to 3.2 at least?

> I guess Quartz.Extensions.Hosting would be logical package name

Makes sense to me! 🙂

I'd be happy to put together a PR for it if you think it's a good idea? 

As an aside, I wasn't sure whether this sort of question is best raised as a discussion or an issue? Discussion feels less pushy when it's an option, hence raising it here! 🙂

**Reply by lahma** [reply]
Yes using the new discussion feature feels natural at least for me for starting the convo and brainstorming. I've created separate issue https://github.com/quartznet/quartznet/issues/911 for getting this done. A PR would be more than welcome ❤️ 

I think pushing Quartz.AspNetCore to be more specific to web world is the way forward and offering hosting separate like you've described. AspNetCore could possibly include also the planned HTTP API bits or maybe even some rudimentary control / status UI.

**Reply by andrewlock** [reply]
Brilliant, I'll aim to get that PR for you soon 🙂 

> I think pushing Quartz.AspNetCore to be more specific to web world is the way forward and offering hosting separate like you've described. AspNetCore could possibly include also the planned HTTP API bits or maybe even some rudimentary control / status UI.

Sounds great to me!

Thanks for all your work on Quartz it's really appreciated 🙂
